File: lib/src/plugin/fair_common_plugin.dart
import 'package:fair/fair.dart';
import 'package:fair_gallery/src/plugin/debug.dart';
import 'package:fair_gallery/src/plugin/future.dart';
import 'package:fair_gallery/src/plugin/http.dart';
import 'package:fair_gallery/src/plugin/launch_url.dart';
import 'package:fair_gallery/src/plugin/navigator.dart';
import 'package:fair_gallery/src/plugin/photo.dart';
import 'package:fair_gallery/src/plugin/totast.dart';

class FairCommonPlugin extends IFairPlugin
    with
        NavigatorPlugin,
        PhotoPlugin,
        ToastPlugin,
        LaunchUrlPlugin,
        CompleterPlugin,
        HttpPlugin,
        FuturePlugin,
        DebugPlugin {
  factory FairCommonPlugin() => _fairCommonPlugin;
  FairCommonPlugin._();
  static final FairCommonPlugin _fairCommonPlugin = FairCommonPlugin._();
  @override
  Map<String, Function> getRegisterMethods() {
    return <String, Function>{
      'savePhoto': savePhoto,
      'navigate': navigate,
      'showToast': showToast,
      'launchUrl': launchUrl,
      'futureComplete': futureComplete,
      'http': http,
      'futureDelayed': futureDelayed,
      'jsPrint': jsPrint,
    };
  }
}

File: assets/plugin/fair_common_plugin.js
// 由 bin/fair_common_plugin.dart 生成
let FairCommonPlugin = function () {
    return {
        futureComplete: function (resp) {
             fairCommonPluginRequest(resp, 'futureComplete');
        },
        launchUrl: function (resp) {
             fairCommonPluginRequest(resp, 'launchUrl');
        },
        savePhoto: function (resp) {
             fairCommonPluginRequest(resp, 'savePhoto');
        },
        futureDelayed: function (resp) {
             fairCommonPluginRequest(resp, 'futureDelayed');
        },
        navigate: function (resp) {
             fairCommonPluginRequest(resp, 'navigate');
        },
        showToast: function (resp) {
             fairCommonPluginRequest(resp, 'showToast');
        },
        jsPrint: function (resp) {
             fairCommonPluginRequest(resp, 'jsPrint');
        },
        http: function (resp) {
             fairCommonPluginRequest(resp, 'http');
        }                     
    }
}
